DeBroglie/Constraints/MaxConsecutiveTilesConstraint.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Collections.Generic;
+
+namespace DeBroglie.Constraints
+{
+    /// <summary>
+    /// Restrict how many consecutive tiles can contain the target tile on one 
+    /// row of an axis. For instance, you might use this constraint to limit how 
+    /// long a ladder can be in a platformer.
+    /// </summary>
+    public class MaxConsecutiveTilesConstraint : ITileConstraint
+    {
+        private int[] lengths;
+        private int beforeDirection;
+        private int afterDirection;
+
+        public Tile Tile { get; set; }
```

Comment:
Like PathConstraint, this is better off accepting a set of tiles.

Reply:
Yea easy enough change. On that note should FixedTileConstraint have the same treatment? In a use case I have I may have a hundred fixed tiles, which would mean creating one constraint per tile.